lib/cli/kit/submodule_loader.rb

```diff
@@ -0,0 +1,42 @@
+require 'pathname'
+require 'cli/kit/util'
+
+# Utility to register submodules or classes with autoload,
+# if they follow the pattern of Foo::PascalCase => foo/snake_case.rb
+#
+# Usage:
+#
+# /path/to/lib/app.rb
+# module App
+#   extend CLI::Kit::SubmoduleLoader
+#   autoload_submodule :Foo        # /path/to/lib/app/foo.rb
+# end
+#
+# /path/to/lib/app/foo.rb
+# module App
+#   module Foo
+#     extend CLI::Kit::SubmoduleLoader
+#     autoload_submodule :Bar      # /path/to/lib/app/foo/bar.rb
+#     autoload_submodule :Baz      # /path/to/lib/app/foo/baz.rb
+#   end
+# end
+module CLI
+  module Kit
+    module SubmoduleLoader
```

Comment:
I didn't feel the need to add specific tests for this since it's tested while loading all classes.